api/models.go

```diff
@@ -166,16 +165,31 @@ type Transaction struct {
 	YParity             *hexutil.Uint64          `json:"yParity,omitempty"`
 }
 
-func NewTransaction(
+func NewTransactionResult(
```

Comment:
changes in this file are a bit unrelated refactoring (because I needed new transaction result only), but improved some other stuff here as well